electrode_models/nucleation_single_particle/initialize.py

```diff
@@ -0,0 +1,129 @@
+        SV_offset = offset
+
+        # Determine the electrode capacity (Ah/m2)
+
+        # Max concentration of stored ion (intercalated Li)
+        # Save initial X
+        X_o = bulk_obj.X 
+        # Set object concentration to fully lithiated:
+        bulk_obj.X = inputs['stored-ion']['name']+':1.0' 
+        # Concentration of stored Li, per unit volume of intercalation phase:
+        C = bulk_obj[inputs['stored-ion']['name']].concentrations[0]
+        
+        capacity = (C*inputs['stored-ion']['charge']*ct.faraday
+                *inputs['eps_solid'])*inputs['thickness']/3600
+
+        # Return Cantera object composition to original value:
+        bulk_obj.X = X_o
+        
+        # Number of state variables: electrode potential, electrolyte 
+        # potential, electrode composition (n_species), electrolyte composition 
+        # (n_species)
+        nVars = 2 + bulk_obj.n_species + elyte_obj.n_species + inputs['buckets']
+
+        # Load the residual function and other required functions and store 
+        # them as methods of this class:
+        from .functions import residual, voltage_lim
+
+    # Set Cantera object state:
+    electrode.bulk_obj.electric_potential = inputs['phi_0']
+    # If the user provided an initial composition, use that, here:
+    if 'X_0' in inputs:
+        electrode.bulk_obj.TPX = (params['T'], params['P'], inputs['X_0'])
+    else:
+        electrode.bulk_obj.TP = params['T'], params['P']
+
+    electrode.elyte_obj.TP = params['T'], params['P']
+    electrode.surf_obj.TP = params['T'], params['P']
+    electrode.conductor_obj.TP = params['T'], params['P']
+
+    # Initialize the solution vector for the electrode domain:
+    SV = np.zeros([electrode.nVars])
+
+    # Set up pointers to specific variables in the solution vector:
+    electrode.SVptr = {}
+    electrode.SVptr['phi_ed'] = np.array([0])
+    electrode.SVptr['phi_dl'] = np.array([1])
+    electrode.SVptr['C_k_ed'] = np.arange(2, 2 + electrode.bulk_obj.n_species)
+    electrode.SVptr['C_k_elyte'] = np.arange(2 + electrode.bulk_obj.n_species, 
+        2 + electrode.bulk_obj.n_species + electrode.elyte_obj.n_species)
+
+    # A pointer to where the SV varaibles for this electrode are, within the 
+    # overall solution vector for the entire problem:
+    electrode.SVptr['residual'] = np.arange(offset, offset+electrode.nVars)
+
+
+    # Save the indices of any algebraic variables:
+    electrode.algvars = offset + electrode.SVptr['phi_ed'][:]
+    electrode.radius = np.linspace(inputs['Minrad'], inputs['Maxrad'], inputs['buckets'])
+
+    # Load intial state variables:
+    SV[electrode.SVptr['phi_ed']] = inputs['phi_0']
+    SV[electrode.SVptr['phi_dl']] = sep_inputs['phi_0'] - inputs['phi_0']
+    SV[electrode.SVptr['C_k_ed']] = electrode.bulk_obj.concentrations
+    SV[electrode.SVptr['C_k_elyte']] = electrode.elyte_obj.concentrations
+    SV[electrode.SVptr['Histogram']] = np.zeros_like(electrode.radius)
```

Comment:
This line probably errors right now, since `SVptr['Histogram']` is not defined. Correct?  

Also, `electrode.radius` currently gives the bin limits (the first value is `Minrad`, the last is `Maxrad`.  If so, I think the buckets actually exist _between_ the elements of `electrode.radius`, so you actually have one fewer bucket. than that given in `inputs[buckets]`. If you want to keep it this way, you can replace with:

`SV[electrode.SVptr['Histogram']] = np.zeros_like(electrode.radius[:-1])`